### ISSUE TITLE
HTML5 .get_data() on Texture returns null

### ISSUE THREAD

**Issue by rokasv** [godotengine/godot#19789]
Godot 3.0.2.stable (also on 3.0.4)

Export this to HTML5:
```
var ss = load("res://spritesheet.png")
var ssdata = ss.get_data()
```

`ss` prints `[StreamTexture]` as expected
`ssdata` is `null` and game is kaput

**Comment by volzhs** [comment]
it seems the same issue with #16694

**Comment by lethiandev** [comment]
It's a common issue on OpenGL ES, because embedded OpenGL does not allow bidirectional data access via glGetTexImage. But it's the problem only with the textures, try using Image's API directly:

```gdscript
var img = Image.new()
img.load("res://icon.png")
var imgdata = img.get_data()
```

Remember to lock the image before data manipulating.

**Comment by rokasv** [comment]
That shifts the error to:
```
**ERROR**: Error opening file: res://Sprites/ss2.png
**ERROR**:    At: core/io/image_loader.cpp:55:load_image() - Error opening file: res://Sprites/ss2.png
```
(It's all good on desktop as it was previously)

I guess it's a different issue, but they might be related enough to keep it here.

**Comment by lethiandev** [comment]
Godot stores all imported assets under the `.import` directory. Try open the ini-like file `res://Sprites/ss2.png.import` and fetch for the location of the image.

i.e.:
```gdscript
var cfg = ConfigFile.new()
cfg.load("res://Sprites/ss2.png.import")

var img = Image.new()
var srcpath = cfg.get_value("deps", "dest_files")[0]
img.load(srcpath)
```

**Comment by rokasv** [comment]
That's cool info, though while the path is returned fine and the file exists, the img data is empty after loading. This is on the desktop.

**Comment by lethiandev** [comment]
Seems that Image's API can't load the streamed textures. Try reading the file manually using File's API.

Set the import flag to `uncompressed` for the texture and read the file:
```gdscript
var cfg = ConfigFile.new()
cfg.load("res://Sprites/ss2.png.import")
print(cfg.get_value("remap", "path"))

var srcpath = cfg.get_value("remap", "path")

var fin = File.new()
fin.open(srcpath, File.READ)

var header = fin.get_32();
var width = fin.get_32();
var height = fin.get_32();
var flags = fin.get_32();
var data_format = fin.get_32();

print("w: ", width, " h:", height)

# assume it's RGBA8 format:
var buffsize = width * height * 4;
var imgdata = fin.get_buffer(buffsize);

# copy the data to Image object
var img = Image.new()
img.create_from_data(width, height, false, Image.FORMAT_RGBA8, imgdata)

# create texture from image
var t = ImageTexture.new()
t.create_from_image(img)
```

**Comment by rokasv** [comment]
Where do I set the import flag?

**Comment by lethiandev** [comment]
On the right panel under _Import_ tab:
![image](https://user-images.githubusercontent.com/5323129/41944216-c14eebc0-799e-11e8-9122-eaceeb1cd7a2.png)
Set _Mode_ flag to _Uncompressed_ and click the _Reimport_ button.

**Comment by rokasv** [comment]
Is there a node that you have the texture assigned to that enables this window?

![capture](https://user-images.githubusercontent.com/13801432/41944304-2067677c-799f-11e8-9052-96ed8e9539d8.PNG)

**Comment by lethiandev** [comment]
Click on the resource on the FileSystem tab, the options should appear.

**Comment by rokasv** [comment]
That did it! 
Thank you for the patience.

**Comment by lethiandev** [comment]
You welcome, but remember that's a hacky way of doing this and requires the every loaded image to be uncompressed, which may result in a bigger .pck file.

**Comment by rokasv** [comment]
It wouldn't be gamedev without the many needed hacks. 

Either way, I imagine this whole thread is enough to patch the issue properly.

**Comment by RKelln** [comment]
I'm also seeing the problem (StreamTexture not returning an image when using `get_data()`) on Android. Using the hack does work, but results in some of the pngs being loaded correctly and others with artifacts/distortions.

**Comment by lethiandev** [comment]
Propably you have mixed data formats, the solution what i presented works only with uncompressed images in RGBA8 format. Try saving all images in png format and ensure you've selected the RGBA8 data format on save.

If you want to, you could also write some checks for the data_format and load the image properly.

**Comment by lethiandev** [comment]
It's an internal limit of the device with getting the texture data back from the GPU. Androids uses OpenGl ES, browsers' WebGL as well.

**Comment by eska014** [comment]
Fixed with #23125
